vsintegration/src/FSharp.Editor/Diagnostics/DocumentDiagnosticAnalyzer.fs

```diff
@@ -125,14 +125,14 @@ type internal FSharpDocumentDiagnosticAnalyzer
             |> RoslynHelpers.StartAsyncAsTask cancellationToken
 
         member this.AnalyzeSemanticsAsync(document: Document, cancellationToken: CancellationToken): Task<ImmutableArray<Diagnostic>> =
+            if document.Project.IsFSharpMiscellaneousOrMetadata && not document.IsFSharpScript then Task.FromResult(ImmutableArray.Empty)
```

Comment:
Typically here we would move this into the computation expression and say

```fsharp
do! (not document.Project.IsFSharpMiscellaneousOrMetadata || document.IsFSharpScript)
```

Reply:
It does, it's `do! Option.guard expr` where expr is a `bool`-returning thing. This is the canonical way that invariants are held in this layer of the tools. I'll submit a separate PR for this, since we should be consistent